Flesh out README, add release schedule (#127)
open-telemetry/opentelemetry-python: README.md

## Installation

This repo include multiple installable packages. The `opentelemetry-api`
package includes abstract classes that comprise the OpenTelemetry API following
[the

> specification](https://github.com/open-telemetry/opentelemetry-specification).

Author: Added in 45b615e.

The `opentelemetry-sdk` package is the reference implementation of the API.

> To install the API and SDK packages, fork or clone this repo and do an

Author: I thought about adding this, but there's a lot of benefit to keeping the readme short and... readable. Most people won't need this -- I think it's fine to leave other installation options as an exercise for the reader.

> To install the API and SDK packages, fork or clone this repo and do an

Reviewer: I agree that keeping readme short and readable is a higher priority.
Maybe later we could explore similar thing here https://github.com/census-instrumentation/opencensus-python/pull/549/files.

> [editable

Reviewer 2: Why do we recommend the editable install?

Author: Only because we haven't pushed anything to PyPI yet. I don't have a strong opinion on this vs. `python setup.py install`, but it looks like there are [a few good reasons](https://stackoverflow.com/questions/30306099/pip-install-editable-vs-python-setup-py-develop) to prefer `pip` here.

install](https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs)
via `pip`:

```sh
pip install -e ./opentelemetry-api
pip install -e ./opentelemetry-sdk
```

The
[`ext/`](https://github.com/open-telemetry/opentelemetry-python/tree/master/ext)
directory includes OpenTelemetry integration packages, which can be installed
separately as:

```sh
pip install -e ./ext/opentelemetry-ext-{integration}
```

> ## Development

Reviewer 3: I feel like this should move to contributing.md. But maybe this heading should exist, pointing to contributing.md again.

Reviewer 2: If we want a "Development section here at all, it should be a subsection of Contributing (or at least come after that section).

Author: Moved in 31723bc.

## Contributing

See [CONTRIBUTING.md](CONTRIBUTING.md)

## Release Schedule

OpenTelemetry Python is under active development. Our goal is to release an
_alpha_ version of the library at the end of September 2019. This release isn't
guaranteed to conform to a specific version of the specification, and is likely

> to include backwards-incompatible changes in later releases.

Reviewer 2: ```suggestion
that backwards-incompatible changes are introduced in later releases.
```

We should also mention that the alpha will only be supported until the beta comes out/not supported at all.

Author: That's a good thing to point out, changed in 99e6f9a.

| Component                   | Version | Target Date       |
| --------------------------- | ------- | ----------------- |
| Tracing API                 | Alpha   | September 30 2019 |
| Tracing SDK                 | Alpha   | September 30 2019 |
| Metrics API                 | Alpha   | September 30 2019 |
| Metrics SDK                 | Alpha   | September 30 2019 |
| Jaeger Trace Exporter       | Alpha   | Unknown           |
| Prometheus Metrics Exporter | Alpha   | Unknown           |
| Context Propagation         | Alpha   | September 30 2019 |

> | OpenTracing Bridge          | Alpha   | Unknown           |

Reviewer: IIRC OpenTracing Bridge for Python has targeted 9/30, please help to double check.

Reviewer 2: In the [Python SIG meeting of 2019-08-29](https://docs.google.com/document/d/1CIMGoIOZ-c3-igzbd6_Pnxx1SjAkjwqoYSUWxPY8XIs/edit#heading=h.rsk1uqixrfg0) with @tedsuo we said that we can't promise anything for the bridges yet because they requirements for them are underspecified. I think the remaining goals are ambitious enough (considering e.g. the current state of the context propagation spec discussion).

Author: I don't know that we need the OpenTracing bridge for the alpha release, and it's unlikely that we'd get the OC bridge done by then. If it looks like we'll get it out in time we can update the date, here or in another PR.

> | OpenCensus Bridge           | Alpha   | Unknown           |

Reviewer 3: is it worth covering propagation/formatters supported? We have B3 header support now, and I believe @a-feld is working on tracestate.

Reviewer 2: That's why I think that the bridges are underspecified. A minimal bridge IMHO only needs to support instrumenting code, i.e. is completely independent of concrete propagators/formatters. But of course, a more complete compatibility would also imply support for all propagators and exporters that OC supports, including OC's own protocol (100% compat would even need to support plugging existing OC exporters into opentelemetry-python, but I don't think we are aiming for that). Which reminds me that the OC bridge will be somewhat trickier than the OT bridge because since there is no API layer, we will probably need to occupy the same package names that OC does to support migration without source modification?

Author: These topics mostly come from @tedsuo's release doc: https://docs.google.com/spreadsheets/d/1FHt3uJgQWN-UWbgV4uMIPUr36JiJCfZ7ryQbHfurjKA/edit#gid=0. There's a checklist section at the bottom that includes the Jaeger and Prometheus exporters, but I didn't include them here. If we want to commit to including them in the release we can add them.

I didn't include either bridge, but we may want to include at least the OpenTracing bridge here too and leave OC for a later release.
